pokemongo_bot/cell_workers/pokemon_catch_worker.py

```diff
@@ -405,8 +405,7 @@ def _do_catch(self, pokemon, encounter_id, catch_rate_by_ball, is_vip=False):
                 )
 
                 # We could refresh here too, but adding 3 saves a inventory request
-                candy = inventory.candies().get(pokemon.num)
-                candy.add(3)
+                candy = inventory.candies(True).get(pokemon.num)
```

Comment:
The refresh is not for the candy, but for the Pokemon. If this refresh is not done, there is nobody in the loop accounting for the -1 space in the Pokemon bag. It leads to inventory full.
I agree that we should be able to do that without a refresh but we have to be careful since it would mean desynchronizing the space left in the bag with the actually list of Pokemons return by inventory.pokemons.
I propose we create a counter in Pokemons inventory class, that we can increase/decrease just like candies.
But if someone call inventory.pokemons() and the size of that list is different from the counter, then we should immediately refresh.